pyanaconda/modules/payloads/source/harddrive/initialization.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os.path
+
+from pyanaconda.core.storage import device_matches
+
+from pyanaconda.modules.common.errors.payload import SourceSetupError
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.errors.storage import MountFilesystemError, DeviceSetupError
+
+from pyanaconda.payload.utils import mount, unmount, PayloadSetupError
+from pyanaconda.payload.image import verify_valid_installtree, find_first_iso_image
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SetUpHardDriveSourceTask", "TearDownHardDriveSourceTask"]
+
+
+class TearDownHardDriveSourceTask(Task):
+    """Task to teardown installation source."""
+
+    def __init__(self, target_mount):
+        super().__init__()
+        self._device_mount = target_mount + "_device"
+        self._iso_mount = target_mount + "_iso"
+
+    @property
+    def name(self):
+        return "Tear down Hard drive installation source"
+
+    def run(self):
+        """Run live installation source un-setup."""
+        log.debug("Unmounting Hard drive installation source")
+        unmount(self._device_mount)
+        unmount(self._iso_mount)
+
+
+class SetUpHardDriveSourceTask(Task):
+    """Task to setup installation source."""
+
+    def __init__(self, target_mount, partition, directory):
+        # TODO need better name for target_mount
+        super().__init__()
+        self._device_mount = target_mount + "_device"
+        self._iso_mount = target_mount + "_iso"
```

Comment:
Maybe these mount points could be also created in the module, so the tasks could get `device_mount` and `iso_mount` instead of the `target_mount`. The best practice is to create them only once.